test/unit/product/product-component.js

```diff
@@ -501,6 +568,106 @@ describe('Product Component class', () => {
       });
     });
 
+    describe('onBlockButtonKeyup()', () => {
+      let event;
+      let onButtonClickStub;
+      const target = {};
+
+      beforeEach(() => {
+        event = {};
+        onButtonClickStub = sinon.stub(product, 'onButtonClick');
+      });
+
+      afterEach(() => {
+        onButtonClickStub.restore();
+      });
+
+      it('calls onButtonClick if event keycode is the enter key', () => {
+        event.keyCode = 13; // enter key
+        product.onBlockButtonKeyup(event, target);
+        assert.calledOnce(onButtonClickStub);
+        assert.calledWith(onButtonClickStub, event, target);
+      });
+
+      it('does not call onButtonClick if event keycode is not the enter key', () => {
+        event.keyCode = 99;
+        product.onBlockButtonKeyup(event, target);
+        assert.notCalled(onButtonClickStub);
+      });
+    });
+
+    describe('onOptionSelect()', () => {
+      it('calls updateVariant with event target name and value', () => {
+        const updateVariantStub = sinon.stub(product, 'updateVariant');
+        const event = {
+          target: {
+            getAttribute(arg) {
+              return arg;
+            },
+            selectedIndex: 'index',
+            options: {
+              index: {
+                value: 'targetValue',
+              },
+            },
+          },
+        };
+        product.onOptionSelect(event);
+        assert.calledOnce(updateVariantStub);
+        assert.calledWith(updateVariantStub, 'name', 'targetValue');
+        updateVariantStub.restore();
+      });
+    });
+
+    describe('onQuantityBlur()', () => {
+      it('calls updateQuantity with function to parse target value', () => {
+        const target = {
+          value: '10',
+        };
+        const event = {};
+        const updateQuantityStub = sinon.stub(product, 'updateQuantity');
+        product.onQuantityBlur(event, target);
+        assert.calledOnce(updateQuantityStub);
+        const value = updateQuantityStub.getCall(0).args[0]();
```

Comment:
Can we name this to be something like `expectedCallbackValue`? It's a bit more clear of what it actually is...